code/modules/antagonists/ninja/energy_net_nets.dm

```diff
@@ -24,22 +23,36 @@
 	buckle_lying = 0
 	buckle_prevents_pull = TRUE
 	///The creature currently caught in the net
-	var/mob/living/affected_mob
+	var/datum/weakref/affected_mob
 
 /obj/structure/energy_net/play_attack_sound(damage, damage_type = BRUTE, damage_flag = 0)
 	if(damage_type == BRUTE || damage_type == BURN)
 		playsound(src, 'sound/weapons/slash.ogg', 80, TRUE)
 
 /obj/structure/energy_net/Destroy()
-	if(!QDELETED(affected_mob))
-		affected_mob.visible_message(span_notice("[affected_mob.name] is recovered from the energy net!"), span_notice("You are recovered from the energy net!"), span_hear("You hear a grunt."))
 	affected_mob = null
 	return ..()
 
+/obj/structure/energy_net/atom_destruction(damage_flag)
+	var/mob/recovered_mob = affected_mob?.resolve()
+	if(recovered_mob)
+		recovered_mob.visible_message(span_notice("[recovered_mob] is recovered from the energy net!"), span_notice("You are recovered from the energy net!"), span_hear("You hear a grunt."))
+	return ..()
```

Comment:
I feel like we can skip out the middle man and just send this message from the buckled list, IE `for(var/mob/buckled as anything in buckled_mobs) buckled.visible_message(...)`

Also maybe should be in `deconstruct` rather than `atom_destruction` though I don't think it matters too much